code/modules/projectiles/gun_system.dm

```diff
@@ -803,18 +823,23 @@ and you're good to go.
 		flick("[fire_animation]", src)
 
 	last_fired = world.time
+
+	return TRUE
+
+///Performs the after firing functions.
+/obj/item/weapon/gun/proc/finish_fire()
```

Comment:
why are you spliting the proc? This is byond, what is good practice in other language is not good here